backend/app/api/routers/graph.py

```
@@ -0,0 +1,428 @@
+        )
+    except Exception as exc:
+        LOGGER.error("Failed to store entities: %s", exc)
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail={"error": "Failed to store entities", "message": str(exc)},
+        )
+
+
+@router.post("/{document_id}/relationships", response_model=Dict[str, Any])
+async def discover_relationships(
+    document_id: str, request: RelationshipDiscoveryRequest
```

Ruff (0.14.10)

Unused function argument: `request`

(ARG001)